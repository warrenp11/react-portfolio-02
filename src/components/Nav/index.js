import React from "react";

function Nav() {
  const categories = [
    {
      name: "portfolio",
      description: "Links to projects I've created as a web developer",
    },
    { name: "resume", description: "My professional resume" },
  ];

  function categorySelected(name) {
    console.log(`${name} clicked`);
  }

  return (
    <header>
      <h1>
        <a data-testid="link" href="/">Hi, my name is Peter!</a>
      </h1>
      <nav>
        <ul className="flex-row">
          <li className="mx-2">
            <a data-testid="about" href="#about">About me</a>
          </li>
          <li>
            <span>Contact</span>
          </li>
          {categories.map((category) => (
            <li className="mx-1" key={category.name}>
              <span onClick={() => categorySelected(category.name)} >{category.name}</span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

export default Nav;
