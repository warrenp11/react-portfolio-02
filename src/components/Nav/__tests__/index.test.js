import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

const categories = [
  { name: "projects", description: "Just a few of the projects I've created as a web developer" }
];
const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();
const mockContactSelected = jest.fn();
const mockSetContactSelected = jest.fn();

afterEach(cleanup);

describe("Nav component", () => {
  it("renders", () => {
    render(<Nav
    categories={categories}
    setCurrentCategory={mockSetCurrentCategory}
    currentCategory={mockCurrentCategory}
    contactSelected={mockContactSelected}
    setContactSelected={mockSetContactSelected}
  />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Nav
    categories={categories}
    setCurrentCategory={mockSetCurrentCategory}
    currentCategory={mockCurrentCategory}
    contactSelected={mockContactSelected}
    setContactSelected={mockSetContactSelected}
  />);
    expect(asFragment()).toMatchSnapshot();
  });
});

// Test for Link Visibility
describe("links are visible", () => {
  it("inserts text into the links", () => {
    // Arrange
    const { getByTestId } = render(<Nav
    categories={categories}
    setCurrentCategory={mockSetCurrentCategory}
    currentCategory={mockCurrentCategory}
    contactSelected={mockContactSelected}
    setContactSelected={mockSetContactSelected}
  />);
    // Assert
    expect(getByTestId("link")).toHaveTextContent("Hi, my name is Peter!");
    expect(getByTestId("about")).toHaveTextContent("About Me");
  });
});