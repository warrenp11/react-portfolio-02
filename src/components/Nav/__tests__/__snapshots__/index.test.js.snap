// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1>
      <a
        data-testid="link"
        href="/"
      >
        Hi, my name is Peter!
      </a>
    </h1>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About Me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1 false"
        >
          <span>
            Projects
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
